src/shared/ListRepo/ReposTable/ReposTable.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import cs from 'classnames'
+import isEmpty from 'lodash/isEmpty'
+import { useContext, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import { OrderingDirection, useRepos } from 'services/repos'
+import { TierNames, useTier } from 'services/tier'
+import { useOwner, useUser } from 'services/user'
+import { ActiveContext } from 'shared/context'
+import Button from 'ui/Button'
+import Icon from 'ui/Icon'
+
+import { getReposColumnsHelper } from './getReposColumnsHelper'
+
+import 'ui/Table/Table.css'
+import { repoDisplayOptions } from '../ListRepo'
+import NoReposBlock from '../NoReposBlock'
+
+interface URLParams {
+  provider: string
+}
+
+interface ReposTableProps {
+  searchValue: string
+  owner: string
+  sortItem?: {
+    ordering?: string
+    direction: string
+  }
+  filterValues?: string[]
+}
+
+function getOrderingDirection(sorting: Array<{ id: string; desc: boolean }>) {
+  const state = sorting.at(0)
+
+  if (state) {
+    const direction = state?.desc
+      ? OrderingDirection.DESC
+      : OrderingDirection.ASC
+
+    let ordering = undefined
+    if (state.id === 'name') {
+      ordering = 'NAME'
+    }
+
+    if (state.id === 'coverage') {
+      ordering = 'COVERAGE'
+    }
+
+    if (state.id === 'latestCommitAt') {
+      ordering = 'COMMIT_DATE'
+    }
+
+    return { direction, ordering }
+  }
+
+  return undefined
+}
+
+const ReposTable = ({
+  searchValue,
+  owner,
+  sortItem,
+  filterValues = [],
+}: ReposTableProps) => {
+  const [sorting, setSorting] = useState<SortingState>([
+    {
+      id: 'latestCommitAt',
+      desc: true,
+    },
+  ])
+
+  const { provider } = useParams<URLParams>()
+
+  const { data: userData } = useUser()
+  const { data: ownerData } = useOwner({
+    username: owner || userData?.user?.username,
+  })
+  const isCurrentUserPartOfOrg = ownerData?.isCurrentUserPartOfOrg
+
+  const { data: tierName } = useTier({ provider, owner })
+  const shouldDisplayPublicReposOnly = tierName === TierNames.TEAM ? true : null
+
+  const repoDisplay = useContext(ActiveContext)
+  const activated =
+    repoDisplayOptions[
+      repoDisplay.toUpperCase() as keyof typeof repoDisplayOptions
+    ]?.status
+
+  const {
+    data: reposData,
+    fetchNextPage,
+    hasNextPage,
+    isFetching,
+    isFetchingNextPage,
+  } = useRepos({
+    activated,
+    sortItem: getOrderingDirection(sorting),
+    term: searchValue,
+    repoNames: filterValues,
+    owner,
+    isPublic: shouldDisplayPublicReposOnly,
+  })
```

Comment:
You'll need to disable suspense in this `useRepos` call, and grab the loading state and render a spinner in the table's body if loading, that way the table headers don't disappear when sorting